### PR TITLE
Update releasing docs from run-through with Nikki

### PR DESCRIPTION
**Pull request by mfisher87**
We discovered an issue: #833 but chose to skip over that problem by giving Nikki temporary (already-revoked) privileges to bypass branch protection. The doc is updated assuming that one can actually push directly to `main`, but I think we should switch to doing all releases in PRs and update the doc accordingly. What does everyone think?

----
📚 Documentation preview 📚: https://earthaccess--834.org.readthedocs.build/en/834/

### CODE REVIEW

**Comment by mfisher87** [comment]
This isn't rendering nicely in Mkdocs. Needs fix!

File: docs/contributing/releasing.md [hunk 1]
```diff
@@ -3,26 +3,46 @@
 > :memo: The versioning scheme we use is [SemVer](http://semver.org/). Note that until
 > we agree we're ready for v1.0.0, we will not increment the major version.
 
-1. Ensure all desired features are merged to `main` branch and `CHANGELOG.md` is updated.
+1. Ensure all desired features are merged to `main` branch and `CHANGELOG.md` is
```
Comment (nikki-t) [inline docs/contributing/releasing.md:6]: Should note that we needed to remove the brackets from `[Unreleased]` in order for `bumpy-my-version` to update the release for the CHANGELOG?

File: docs/contributing/releasing.md [hunk 2]
```diff
@@ -3,26 +3,46 @@
...
+   updated. **Do not** edit the `[Unreleased]` header in the CHANGELOG -- the next step
+   will automatically update it.
 1. Use `bump-my-version` to increase the version number in all needed places, e.g. to
```
Comment (nikki-t) [inline docs/contributing/releasing.md:9]: We should include the files that `bump-my-version` changes which is: CHANGELOG.md, pyproject.toml, and CITATION.cff

**Review by nikki-t — DISMISSED** [review]
I left a few comments on some suggestion but I think the updated documentation on the release process look great!

File: docs/contributing/releasing.md (hunk 1 above)
Comment (mfisher87) [reply docs/contributing/releasing.md:6]: Good catch, the brackets shouldn't have been there in the first place :)

File: docs/contributing/releasing.md [hunk 3]
```diff
@@ -3,26 +3,46 @@
 > :memo: The versioning scheme we use is [SemVer](http://semver.org/). Note that until
 > we agree we're ready for v1.0.0, we will not increment the major version.
 
-1. Ensure all desired features are merged to `main` branch and `CHANGELOG.md` is updated.
+1. Ensure all desired features are merged to `main` branch and `CHANGELOG.md` is
+   updated. **Do not** edit the `[Unreleased]` header in the CHANGELOG -- the next step
```
Comment (mfisher87) [inline docs/contributing/releasing.md:7]:
```suggestion
   updated. **Do not** edit the `Unreleased` header in the CHANGELOG -- the next step
```

File: docs/contributing/releasing.md (hunk 2 above)
Comment (mfisher87) [reply docs/contributing/releasing.md:9]: I'm not sure how I feel about that. I could be convinced either way. The other side, in my mind, is then we have to keep that list in sync when we change the config. Perhaps something along the lines of "The files which will be edited by bump-my-version are configured in `pyproject.toml`"?

Comment (nikki-t) [reply docs/contributing/releasing.md:9]: That makes sense. I like the idea of pointing folks to a list of files that will be edited (`pyproject.toml`) as it also provides some context for the `bump-my-version` command.

Comment (mfisher87) [reply docs/contributing/releasing.md:9]: 3004088 What do you think?

**Comment by mfisher87** [comment]
Fixed up the formatting in Mkdocs! 5b84afc

File: docs/contributing/releasing.md (hunk 2 above)
Comment (nikki-t) [reply docs/contributing/releasing.md:10]: Perfect!

**Comment by nikki-t** [comment]
Looks great!

**Review by nikki-t — APPROVED** [review]

**Comment by mfisher87** [comment]
Thanks, @nikki-t !

**Pull request merged**